zips/zip-0312.rst

```diff
@@ -163,44 +159,68 @@ following modifications.
 Randomizer Generation
 '''''''''''''''''''''
 
-A new helper function is defined, which generates a randomizer. The
-`encode_signing_package` is defined as the byte serialization of the `msg`,
-`commitment_list` values as described in [#frost-serialization]_.
-Implementations MAY choose another encoding as long as all values (the message,
-and the identifier, binding nonce and hiding nonce for each participant) are
+Two new helper functions are defined, which generate a randomizer; the first is
+used by the Coordinator and the second is used by participants. The
+`encode_signing_commitments` function SHOULD return the byte serialization of
+the `commitment_list` value as the serialization of the map between identifiers
+and signing commitments, as described in [#frost-serialization]_.
+Implementations MAY choose another encoding as long as all values (the
+identifier, binding nonce and hiding nonce for each participant) are
 unambiguously encoded.
```

Comment:
There is a risk here that a Coordinator and Participant (or multiple Participants) might have different implementations, and attempt to use them in the same signing session. That would be incompatible and cause a failure, which violates the defintion of MAY in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt):
> 5. MAY   This word, or the adjective "OPTIONAL", mean that an item is
   truly optional.  One vendor may choose to include the item because a
   particular marketplace requires it or because the vendor feels that
   it enhances the product while another vendor may omit the same item.
   An implementation which does not include a particular option MUST be
   prepared to interoperate with another implementation which does
   include the option, though perhaps with reduced functionality. In the
   same vein an implementation which does include a particular option
   MUST be prepared to interoperate with another implementation which
   does not include the option (except, of course, for the feature the
   option provides.)

So we will need to specify the full encoding here.

Reply:
Done in c74bed50 by using the serialization function from the RFC